gengokit/middlewares/middlewares.go

```diff
@@ -0,0 +1,63 @@
+// Package middlewares renders the service and endpointe middleware files in NAME-service/middlewares/.
+package middlewares
+
+import (
+	"io"
+
+	"github.com/pkg/errors"
+
+	"github.com/TuneLab/go-truss/gengokit"
+	"github.com/TuneLab/go-truss/gengokit/middlewares/templates"
+)
+
+// EndpointsPath is the path to the Endpoints middleware file that package middlewares renders
+const EndpointsPath = "NAME-service/middlewares/endpoints.gotemplate"
+
+// ServicePath is the path to the Service middleware file that package middlewares renders
+const ServicePath = "NAME-service/middlewares/service.gotemplate"
+
+// New returns a Middleware which can render EndpointsFile and ServiceFile as
+// well as read in previous versions of each respective file
+func New() *Middlewares {
```

Comment:
If you think that we are going to have functionality that will warrant a `New` method to conceal setup in the `Middlewares` type, then I think it's fine to leave it. I hadn't considered the API symmetry with `handler`, and I don't really know how important it is, but it's not hurting anything, so we can proceed.